ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java

```diff
@@ -150,19 +152,29 @@ public ValidationResult<TransactionInvalidReason> validate(
       }
     }
 
-    // transactionValidationParams.isAllowExceedingBalance() is used on eth_call
-    if (!feeMarket.satisfiesFloorTxCost(transaction)
-        && !transactionValidationParams.isAllowExceedingBalance()) {
+    if (transaction.getNonce() == -1) {
```

Comment:
how is -1 NONCE_TOO_HIGH

Reply:
the max unsigned value is the same as the signed 2's complement -1.  However, I replaced it with MAX_NONCE to make it clear what and why we are testing.